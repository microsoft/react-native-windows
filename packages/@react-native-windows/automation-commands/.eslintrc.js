module.exports = {
    extends: ['@rnw-scripts'],
    parserOptions: {tsconfigRootDir : __dirname},
};
