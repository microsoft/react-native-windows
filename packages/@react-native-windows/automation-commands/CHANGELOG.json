{
  "name": "@react-native-windows/automation-commands",
  "entries": [
    {
      "date": "Fri, 01 Oct 2021 05:05:54 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-2",
      "version": "0.0.6-2",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-2",
            "author": "julio.rocha@microsoft.com",
            "commit": "eaefea114554d0b91dbf2d6aff650b2fc5673f7a",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 30 Sep 2021 05:06:48 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-1",
      "version": "0.0.6-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-1",
            "author": "erozell@outlook.com",
            "commit": "4bc8637395ea828788c0ba267561c1a01f06f484",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 29 Sep 2021 05:06:54 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-0",
      "version": "0.0.6-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-0",
            "author": "anandrag@microsoft.com",
            "commit": "eb04aeb2e5fcb74988f1630f24adc27a034351fe",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 28 Sep 2021 05:07:03 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.5",
      "version": "0.0.5",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.4",
            "author": "34109996+chiaramooney@users.noreply.github.com",
            "commit": "d4970fdd583866c3ac7af6a0cf38040cf3b22d93",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 25 Sep 2021 05:06:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.5-0",
      "version": "0.0.5-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-12",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "eaf932469678b8b67b0b2b17fd2bcdda1ed4e358",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 24 Sep 2021 05:06:42 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4",
      "version": "0.0.4",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-11",
            "author": "igklemen@microsoft.com",
            "commit": "8b63ada9ba1d55bfd161388fc18c2b5c6445b508",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 23 Sep 2021 05:06:48 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4-1",
      "version": "0.0.4-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-10",
            "author": "34109996+chiaramooney@users.noreply.github.com",
            "commit": "3813907b2eaa6198f0a8b9dc570075f0c7117301",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 22 Sep 2021 05:06:43 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4-0",
      "version": "0.0.4-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-9",
            "author": "ali-hk@users.noreply.github.com",
            "commit": "0bc36a30ab3cd5bb4cfd767e503b29c185cfcddd",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 21 Sep 2021 05:08:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3",
      "version": "0.0.3",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-8",
            "author": "igklemen@microsoft.com",
            "commit": "08eb19f42f4f74bd82f33db2e27ba2d33b2ddd1b",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 18 Sep 2021 05:07:45 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-7",
      "version": "0.0.3-7",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-7",
            "author": "erozell@outlook.com",
            "commit": "fbbfb83843c60abb8c02368077fd0a406f0bdcc0",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 17 Sep 2021 05:06:53 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-6",
      "version": "0.0.3-6",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-6",
            "author": "jthysell@microsoft.com",
            "commit": "95fecc02d3dcc9f03d0fc7d47d30cb45e128bca6",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 16 Sep 2021 05:06:44 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-5",
      "version": "0.0.3-5",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-5",
            "author": "erozell@outlook.com",
            "commit": "b46965a75f179c1855a2499c7f3c6b4b10daf416",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 15 Sep 2021 05:07:04 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-4",
      "version": "0.0.3-4",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-4",
            "author": "30809111+acoates-ms@users.noreply.github.com",
            "commit": "5dbd74b1852c38f851483478d6da7d81949d0195",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 14 Sep 2021 05:07:02 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-3",
      "version": "0.0.3-3",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-3",
            "author": "jthysell@microsoft.com",
            "commit": "d3de4a2b04415ff9004b7ba08ee21787b924d0d4",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 11 Sep 2021 05:06:55 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-2",
      "version": "0.0.3-2",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-2",
            "author": "anandrag@microsoft.com",
            "commit": "09bc9974c8721bf5559680dc7a8aa00514c1c7f3",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 10 Sep 2021 05:07:31 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-1",
      "version": "0.0.3-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-1",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "42d0853c684f0e9f9c9f1f7fa34cbf4ca3810edc",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 09 Sep 2021 05:07:46 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-0",
      "version": "0.0.3-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-0",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "66a9f86f24f862c9e9e611ad435f68daddb4e04e",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 08 Sep 2021 05:08:53 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.2",
      "version": "0.0.2",
      "comments": {
        "patch": [
          {
            "comment": "Set consistent node requirements on our packages",
            "author": "ngerlem@microsoft.com",
            "commit": "2974ea0ab58b546264b8d9a4a7c12ceeb0a02851",
            "package": "@react-native-windows/automation-commands"
          }
        ],
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.2",
            "author": "ngerlem@microsoft.com",
            "commit": "2974ea0ab58b546264b8d9a4a7c12ceeb0a02851",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 04 Sep 2021 05:06:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.2-0",
      "version": "0.0.2-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.2-0",
            "author": "dannyvv@microsoft.com",
            "commit": "d0c3465f27bd0de739b23b46e957cbdab6bdf014",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 03 Sep 2021 05:06:04 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.1",
      "version": "0.0.1",
      "comments": {
        "patch": [
          {
            "comment": "@react-native-windows/automation",
            "author": "ngerlem@microsoft.com",
            "commit": "17e32d36180fa5c7e45c3c926c6eaa729efb6ea0",
            "package": "@react-native-windows/automation-commands"
          }
        ],
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.1",
            "author": "erozell@outlook.com",
            "commit": "439b8a54b62807068bfa17e130f8ab0378eaaf36",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    }
  ]
}
