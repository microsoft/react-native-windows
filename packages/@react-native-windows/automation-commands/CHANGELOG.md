# Change Log - @react-native-windows/automation-commands

This log was last generated on Fri, 01 Oct 2021 05:05:54 GMT and should not be manually modified.

<!-- Start content -->

## 0.0.6-2

Fri, 01 Oct 2021 05:05:54 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.5-2 (julio.rocha@microsoft.com)

## 0.0.6-1

Thu, 30 Sep 2021 05:06:48 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.5-1 (erozell@outlook.com)

## 0.0.6-0

Wed, 29 Sep 2021 05:06:54 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.5-0 (anandrag@microsoft.com)

## 0.0.5

Tue, 28 Sep 2021 05:07:03 GMT

### Patches

- Bump @react-native-windows/automation-channel to v0.0.4 (34109996+chiaramooney@users.noreply.github.com)

## 0.0.5-0

Sat, 25 Sep 2021 05:06:39 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-12 (53799235+ZihanChen-MSFT@users.noreply.github.com)

## 0.0.4

Fri, 24 Sep 2021 05:06:42 GMT

### Patches

- Bump @react-native-windows/automation-channel to v0.0.3-11 (igklemen@microsoft.com)

## 0.0.4-1

Thu, 23 Sep 2021 05:06:48 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-10 (34109996+chiaramooney@users.noreply.github.com)

## 0.0.4-0

Wed, 22 Sep 2021 05:06:43 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-9 (ali-hk@users.noreply.github.com)

## 0.0.3

Tue, 21 Sep 2021 05:08:39 GMT

### Patches

- Bump @react-native-windows/automation-channel to v0.0.3-8 (igklemen@microsoft.com)

## 0.0.3-7

Sat, 18 Sep 2021 05:07:45 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-7 (erozell@outlook.com)

## 0.0.3-6

Fri, 17 Sep 2021 05:06:53 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-6 (jthysell@microsoft.com)

## 0.0.3-5

Thu, 16 Sep 2021 05:06:44 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-5 (erozell@outlook.com)

## 0.0.3-4

Wed, 15 Sep 2021 05:07:04 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-4 (30809111+acoates-ms@users.noreply.github.com)

## 0.0.3-3

Tue, 14 Sep 2021 05:07:02 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-3 (jthysell@microsoft.com)

## 0.0.3-2

Sat, 11 Sep 2021 05:06:55 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-2 (anandrag@microsoft.com)

## 0.0.3-1

Fri, 10 Sep 2021 05:07:31 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-1 (53799235+ZihanChen-MSFT@users.noreply.github.com)

## 0.0.3-0

Thu, 09 Sep 2021 05:07:46 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.3-0 (53799235+ZihanChen-MSFT@users.noreply.github.com)

## 0.0.2

Wed, 08 Sep 2021 05:08:53 GMT

### Patches

- Set consistent node requirements on our packages (ngerlem@microsoft.com)

### Changes

- Bump @react-native-windows/automation-channel to v0.0.2 (ngerlem@microsoft.com)

## 0.0.2-0

Sat, 04 Sep 2021 05:06:39 GMT

### Changes

- Bump @react-native-windows/automation-channel to v0.0.2-0 (dannyvv@microsoft.com)

## 0.0.1

Fri, 03 Sep 2021 05:06:04 GMT

### Patches

- @react-native-windows/automation (ngerlem@microsoft.com)

### Changes

- Bump @react-native-windows/automation-channel to v0.0.1 (erozell@outlook.com)
