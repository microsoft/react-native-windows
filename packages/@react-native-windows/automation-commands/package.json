{
  "name": "@react-native-windows/automation-commands",
  "version": "0.1.31",
  "description": "Allows controlling your react-native-windows application",
  "main": "lib-commonjs/index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git@github.com:microsoft/react-native-windows.git",
    "directory": "packages/@react-native-windows/automation-commands"
  },
  "private": true,
  "scripts": {
    "build": "rnw-scripts build",
    "clean": "rnw-scripts clean",
    "lint": "rnw-scripts lint",
    "lint:fix": "rnw-scripts lint:fix",
    "watch": "rnw-scripts watch"
  },
  "dependencies": {
    "@react-native-windows/automation-channel": "^0.5.11",
    "@typescript-eslint/eslint-plugin": "^5.30.5",
    "@typescript-eslint/parser": "^5.30.5"
  },
  "devDependencies": {
    "@jest/types": "^29.2.1",
    "@rnw-scripts/eslint-config": "1.1.14",
    "@rnw-scripts/just-task": "2.3.3",
    "@rnw-scripts/ts-config": "2.0.2",
    "@types/jest": "^29.2.2",
    "@types/node": "^14.14.22",
    "eslint": "^8.19.0",
    "prettier": "^2.4.1",
    "typescript": "^4.9.5"
  },
  "files": [
    "lib-commonjs",
    "README.md"
  ],
  "engines": {
    "node": ">= 14"
  }
}