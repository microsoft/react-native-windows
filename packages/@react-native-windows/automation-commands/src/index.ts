/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import dumpVisualTree from './dumpVisualTree';

export {dumpVisualTree};
