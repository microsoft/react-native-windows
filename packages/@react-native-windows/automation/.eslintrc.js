module.exports = {
  extends: ['@rnw-scripts'],
  parserOptions: {tsconfigRootDir : __dirname},
  
  // Workaround "export type" being unsupported with prettier 1.x
  ignorePatterns: ['index.ts'],
  overrides: [
    {
      files: ['*.ts'],
      // Also not clean
      rules: {
        '@typescript-eslint/no-unused-vars': 'off',
      },
    },
  ],
};
