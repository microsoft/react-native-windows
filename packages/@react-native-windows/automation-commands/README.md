# @react-native-windows/automation-commands

`@react-native-windows/automation-commands` provides built-in commands that may
be used with `@react-native-windows/automation-channel` to make it easier to
test your application.

**This package is a work in progress**

## Adding to your application

**TBD**

## Commands

### dumpVisualTree

The `dumpVisualTree` command creates a JSON object corresponding to the XAML
tree under a given testID. This can be used in conjunction with snapshot
testing to validate that your native UI looks how you could expect.

```ts
import {dumpVisualTree} from '@react-native-windows/automation-commands';

test('Widget', async () => {
  const dump = await dumpVisualTree('widget-test-id');
  expect(dump).toMatchSnapshot();
});
```

#### Options
**TBD**