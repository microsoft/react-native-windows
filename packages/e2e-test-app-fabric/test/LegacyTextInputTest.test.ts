/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */
import {app} from '@react-native-windows/automation';
import {goToComponentExample} from './RNTesterNavigation';
import {verifyNoErrorLogs} from './Helpers';
import {dumpVisualTree} from '@react-native-windows/automation-commands';

beforeAll(async () => {
  // If window is partially offscreen, tests will fail to click on certain elements
  await app.setWindowPosition(0, 0);
  await app.setWindowSize(1000, 1250);
  await goToComponentExample('LegacyTextInputTest');
});

afterEach(async () => {
  await verifyNoErrorLogs();
});

describe('LegacyTextInputTest', () => {
  test('Click on TextInput to focus', async () => {
    const textInput = await textInputField();
    await textInput.click();
    await assertLogContains('onFocus');
  });

  test('Click on multiline TextInput to move focus away from single line TextInput', async () => {
    const textInput = await multiLineTextInputField();
    await textInput.click();
    await assertLogContains('onBlur');
  });
  test('Type abc on TextInput', async () => {
    const textInput = await textInputField();
    await textInput.setValue('abc');
    expect(await textInput.getText()).toBe('abc');
    await assertLogContains('onKeyPress key: c');
  });
  test('Type def on TextInput', async () => {
    const textInput = await textInputField();
    await app.waitUntil(
      async () => {
        await textInput.setValue('def');
        return (await textInput.getText()) === 'def';
      },
      {
        interval: 1500,
        timeout: 5000,
        timeoutMsg: `Unable to enter correct text.`,
      },
    );
    expect(await textInput.getText()).toBe('def');
  });
  /* Issue to enable these tests: #12778
  test('Type hello world on autoCap TextInput', async () => {
    const textInput = await autoCapsTextInputField();
    await textInput.setValue('def');
    expect(await textInput.getText()).toBe('DEF');

    await textInput.setValue('hello world');
    expect(await textInput.getText()).toBe('HELLO WORLD');
  });
  */
  test('Type abc on multiline TextInput then press Enter key', async () => {
    const textInput = await textInputField();
    await textInput.setValue('abc');
    await textInput.addValue('Enter');

    await assertLogContains('onSubmitEditing text: abc');
  });
  test('Type abc on multiline TextInput', async () => {
    const textInput = await multiLineTextInputField();
    await textInput.setValue('abc');

    expect(await textInput.getText()).toBe('abc');
  });

  test('Enter key then type def on multiline TextInput', async () => {
    const textInput = await multiLineTextInputField();

    await textInput.addValue('End');
    await textInput.addValue('Enter');
    await textInput.addValue('def');

    expect(await textInput.getText()).toBe('abc\rdef');
  });

  test('TextInput onChange before onSelectionChange', async () => {
    const textInput = await textInputField();
    await textInput.setValue('a');
    await assertLogContains('onChange text: a\nonSelectionChange range: 1,1');
    /* Issue to enable these tests: #12778
    await assertLogContainsInOrder([
      'onChange text: a',
      'onSelectionChange range: 1,1',
    ]);
    */
  });
});

async function textInputField() {
  const component = await app.findElementByTestID('textinput-field');
  await component.waitForDisplayed({timeout: 5000});
  return component;
}

/* Issue to enable these tests: #12778
async function autoCapsTextInputField() {
  const component = await app.findElementByTestID('auto-caps-textinput-field');
  await component.waitForDisplayed({timeout: 5000});
  return component;
}
*/

async function multiLineTextInputField() {
  const component = await app.findElementByTestID('multi-line-textinput-field');
  await component.waitForDisplayed({timeout: 5000});
  return component;
}

async function assertLogContains(_text: string) {
  const textLogComponent = await app.findElementByTestID('textinput-log');
  await textLogComponent.waitForDisplayed({timeout: 5000});

  const dump = await dumpVisualTree('textinput-log');
  expect(dump).toMatchSnapshot();
  /*
  await app.waitUntil(
    async () => {
      const loggedText = await textLogComponent.getText();
      return loggedText.split('\n').includes(text);
    },
    {
      timeoutMsg: `"${await textLogComponent.getValue()}" "${await textLogComponent.getText()}" did not contain "${text}"`,
    },
  );
  */
}

/*
async function assertLogContainsInOrder(expectedLines: string[]) {
  const textLogComponent = await app.findElementByTestID('textinput-log');
  await textLogComponent.waitForDisplayed({timeout: 5000});

  await app.waitUntil(
    async () => {
      const loggedText = await textLogComponent.getText();
      const actualLines = loggedText.split('\n');
      let previousIndex = Number.MAX_VALUE;
      for (const line of expectedLines) {
        const index = actualLines.findIndex(l => l === line);
        if (index === -1 || index > previousIndex) {
          return false;
        }

        previousIndex = index;
      }

      return true;
    },
    {
      timeoutMsg: `"${await textLogComponent.getText()}" did not contain lines "${expectedLines.join(
        ', ',
      )}"`,
    },
  );
}
*/
