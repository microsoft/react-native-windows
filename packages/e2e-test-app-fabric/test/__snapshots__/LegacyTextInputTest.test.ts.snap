// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegacyTextInputTest Click on TextInput to focus 1`] = `
{
  "Automation Tree": {
    "AutomationId": "textinput-log",
    "ControlType": 50020,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": false,
    "LocalizedControlType": "text",
    "Name": "onFocus
<Log Start>",
  },
  "Visual Tree": {
    "Comment": "textinput-log",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      998,
      38,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`LegacyTextInputTest Click on multiline TextInput to move focus away from single line TextInput 1`] = `
{
  "Automation Tree": {
    "AutomationId": "textinput-log",
    "ControlType": 50020,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": false,
    "LocalizedControlType": "text",
    "Name": "onBlur
onFocus
<Log Start>",
  },
  "Visual Tree": {
    "Comment": "textinput-log",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      998,
      57,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`LegacyTextInputTest TextInput onChange before onSelectionChange 1`] = `
{
  "Automation Tree": {
    "AutomationId": "textinput-log",
    "ControlType": 50020,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": false,
    "LocalizedControlType": "text",
    "Name": "onChange text: a
onSelectionChange range: 1,1
onKeyPress key: a
onChange text: 
onSelectionChange range: 0,0
onFocus
onBlur
onSubmitEditing text: abc
onChange text: abc
onSelectionChange range: 3,3
onKeyPress key: c
onChange text: ab
onSelectionChange range: 2,2
onKeyPress key: b
onChange text: a
onSelectionChange range: 1,1
onKeyPress key: a
onChange text: 
onSelectionChange range: 0,0
onChange text: def
onSelectionChange range: 3,3
onKeyPress key: f
onChange text: de
onSelectionChange range: 2,2
onKeyPress key: e
onChange text: d
onSelectionChange range: 1,1
onKeyPress key: d
onChange text: 
onSelectionChange range: 0,0
onChange text: abc
onSelectionChange range: 3,3
onKeyPress key: c
onChange text: ab
onSelectionChange range: 2,2
onKeyPress key: b
onSelectionChange range: 1,1
onChange text: a
onSelectionChange range: 0,0
onChange text: a
onSelectionChange range: 1,1
onKeyPress key: a
onFocus
onBlur
onFocus
<Log Start>",
  },
  "Visual Tree": {
    "Comment": "textinput-log",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      998,
      858,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`LegacyTextInputTest Type abc on TextInput 1`] = `
{
  "Automation Tree": {
    "AutomationId": "textinput-log",
    "ControlType": 50020,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": false,
    "LocalizedControlType": "text",
    "Name": "onChange text: abc
onSelectionChange range: 3,3
onKeyPress key: c
onChange text: ab
onSelectionChange range: 2,2
onKeyPress key: b
onSelectionChange range: 1,1
onChange text: a
onSelectionChange range: 0,0
onChange text: a
onSelectionChange range: 1,1
onKeyPress key: a
onFocus
onBlur
onFocus
<Log Start>",
  },
  "Visual Tree": {
    "Comment": "textinput-log",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      998,
      299,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;

exports[`LegacyTextInputTest Type abc on multiline TextInput then press Enter key 1`] = `
{
  "Automation Tree": {
    "AutomationId": "textinput-log",
    "ControlType": 50020,
    "HelpText": "",
    "IsEnabled": true,
    "IsKeyboardFocusable": false,
    "LocalizedControlType": "text",
    "Name": "onSubmitEditing text: abc
onChange text: abc
onSelectionChange range: 3,3
onKeyPress key: c
onChange text: ab
onSelectionChange range: 2,2
onKeyPress key: b
onChange text: a
onSelectionChange range: 1,1
onKeyPress key: a
onChange text: 
onSelectionChange range: 0,0
onChange text: def
onSelectionChange range: 3,3
onKeyPress key: f
onChange text: de
onSelectionChange range: 2,2
onKeyPress key: e
onChange text: d
onSelectionChange range: 1,1
onKeyPress key: d
onChange text: 
onSelectionChange range: 0,0
onChange text: abc
onSelectionChange range: 3,3
onKeyPress key: c
onChange text: ab
onSelectionChange range: 2,2
onKeyPress key: b
onSelectionChange range: 1,1
onChange text: a
onSelectionChange range: 0,0
onChange text: a
onSelectionChange range: 1,1
onKeyPress key: a
onFocus
onBlur
onFocus
<Log Start>",
  },
  "Visual Tree": {
    "Comment": "textinput-log",
    "Offset": [
      0,
      0,
      0,
    ],
    "Opacity": 1,
    "Size": [
      998,
      727,
    ],
    "Visual Type": "SpriteVisual",
  },
}
`;
